src/lib/make-toolbox-xml.js

```diff
@@ -716,35 +716,57 @@ const xmlClose = '</xml>';
 /**
  * @param {!boolean} isStage - Whether the toolbox is for a stage-type target.
  * @param {?string} targetId - The current editing target
- * @param {?string} categoriesXML - null for default toolbox, or an XML string with <category> elements.
+ * @param {?Array.<object>} categoriesXML - optional array of `{id,xml}` for categories.
```

Comment:
```suggestion
 * @param {?Array.<object>} categoriesXML - optional array of `{id,xml}` for additional ("extension") categories in the order that they should appear in the toolbox. If a category in this list shares an ID with a "core" block category, the XML from this list will be used in place of the default xml defined in this file.
```

Reply:
heh - I wrote another version of this comment at more or less the same time. Let me know if you think mine covers it well enough or if you think I should use yours or merge the two. Thanks :)